gaffer-as-a-service/gaas-jwt-sidecar/src/main/java/uk/gov/gchq/gaffer/gaas/sidecar/controller/CustomFilter.java

```diff
@@ -0,0 +1 @@
import reactor.core.publisher.Mono;
import uk.gov.gchq.gaffer.gaas.sidecar.auth.JwtTokenUtil;
import uk.gov.gchq.gaffer.gaas.sidecar.auth.JwtUserDetailsService;

@Order(1)
@Configuration
public class CustomFilter implements GlobalFilter {

    @Autowired
    private JwtUserDetailsService jwtUserDetailsService;

    @Autowired
    private JwtTokenUtil jwtTokenUtil;

    Logger logger = LoggerFactory.getLogger(CustomFilter.class);

    @Override
    public Mono<Void> filter(final ServerWebExchange exchange, final GatewayFilterChain chain) {

        ServerHttpRequest request = exchange.getRequest();
        if (request.getPath().toString().equals("/auth")) {
            return chain.filter(exchange).then(Mono.fromRunnable(() -> {
                ServerHttpResponse response = exchange.getResponse();
                logger.info("Post Filter = " + response.getStatusCode());
            }));
        }
        final String requestTokenHeader = request.getHeaders().getFirst("Authorization");

        String username = null;
        String jwtToken = null;
        // JWT Token is in the form "Bearer token". Remove Bearer word and get
        // only the Token
        if (requestTokenHeader != null && requestTokenHeader.startsWith("Bearer ")) {
            jwtToken = requestTokenHeader.substring(7);
            try {
                username = jwtTokenUtil.getUsernameFromToken(jwtToken);
            } catch (IllegalArgumentException e) {
                logger.warn("Unable to get JWT Token");
            } catch (ExpiredJwtException e) {
                logger.warn("JWT Token has expired");
            } catch (SignatureException e) {
                logger.warn("JWT validity cannot be asserted");
            }
        } else {
            logger.warn("JWT Token does not begin with Bearer String");
        }

        // Once we get the token validate it.
        if (username != null && SecurityContextHolder.getContext().getAuthentication() == null) {

            UserDetails userDetails = this.jwtUserDetailsService.loadUserByUsername(username);

            // if token is valid configure Spring Security to manually set
            // authentication

            if (jwtTokenUtil.validateToken(jwtToken, userDetails)) {
                return chain.filter(exchange).then(Mono.fromRunnable(() -> {
                    ServerHttpResponse response = exchange.getResponse();
                    logger.info("Post Filter = " + response.getStatusCode());
                }));
            }
        }
        logger.info("Authorization = " + request.getHeaders().getFirst("Authorization"));
        return this.onError(exchange, "Unauthorised User", HttpStatus.FORBIDDEN);
    }

    private Mono<Void> onError(final ServerWebExchange exchange, final String err, final HttpStatus httpStatus) {
        ServerHttpResponse response = exchange.getResponse();
        response.setStatusCode(httpStatus);
        return response.setComplete();
    }
}

```

Comment:
Still strange `^M` line endings, see: https://github.com/gchq/gaffer-experimental/pull/294#discussion_r846070015

Reply:
@t92549 
Not quite sure how to do this?
Have tried following:
https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings#refreshing-a-repository-after-changing-line-endings
Doesn't seem to work. Any tips?

Reply:
You don't need to add a `.gitattributes`, you should just ensure that your editor / local git config don't submit code with CR line endings. To fix already broken files you can just find and replace `\r` with `\n` in the effected files.

Reply:
@t92549 
I've checked my IDE its set to use the Unix style line separator (LF). My local git config is also set to `global core.autocrlf input` I've also tried it with `global core.autocrlf false`. I've run the command `tr '\r' '\n'` on the offending files. None of this seems to be picking up for me. I have also used dos2unix to no avail.
`od -c <file>` shows no '\r` line breaks they are all `\n` so presumably GitHub is doing some sort of conversion on upload as locally the files seem fine although I do think my local git config is correct so not sure what I'm missing.

I have tried to rectify the header dates issue by giving them year of creation - year last edited so hopefully all of the new sidecar related headers are correct now.

Reply:
I've cloned this PR and running `od -c CustomFilter.java | egrep \r` gets lots of results. By default git doesn't transform the line ending, but GitHub and other Unix-like system have trouble when the Windows ones are used. [From the looks of it](https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration#_formatting_and_whitespace), the autocrlf option is not what you want. The `.gitattributes` approach didn't seem to work for me, I think the best method would be to just convert the files with this problem as and when it happens.

I've previously used `cat inputfile | tr '\r' '\n' > outputfile` to fix this - ensure input and output are not the same file!

Reply:
Thank you I think that's worked.